lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -614,12 +612,10 @@ def _get_users(discussion_entity_type, discussion_entity, username_profile_dict)
     """
     users = {}
     if discussion_entity['author']:
-        try:
-            users[discussion_entity['author']] = _user_profile(username_profile_dict[discussion_entity['author']])
-        except ValueError as value_error:
-            log.error(f'User Profile do not exists for Author :{discussion_entity["author"]} '
-                      f'profiles list: {username_profile_dict.keys()} ')
-            raise value_error
+        user_profile = username_profile_dict.get(discussion_entity['author'])
+        if user_profile:
```

Comment:
What is the case when the user profile does not exist?

Reply:
There is a case when the user is retired but the username is referenced in Post as author